avocado/utils/memory.py

```diff
@@ -27,8 +27,75 @@
 from . import genio
 
 
-# Returns total memory in kb
+def _check_memory_state(mem_block):
+    """
+    Check the given memory block is online or offline
+
+    :param mem_block: memory block id.
+    :type string: like 198
+    :return: 'True' if online or 'False' if offline
+    :rtype: bool
+    """
+    if 'online' in open('/sys/devices/system/memory/memory%s/state' % mem_block).read():
+        return True
+    return False
+
+
+def _check_hot_plug():
+    """
+    Check kernel support for memory hotplug
+
+    :return: True if hotplug supported,  else False
+    :rtype: 'bool'
+    """
+    if glob.glob('/sys/devices/system/memory/memory*'):
+        return True
+    return False
+
+
+def _is_hot_pluggable(mem_block):
+    """
+    Check if the given memory block is hotpluggable
+
+    :param mem_block: memory block id.
+    :type string: like 198
+    :retrun: True if hotpluggable, else False
+    :rtype: 'bool'
+    """
+    if os.path.exists('/sys/devices/system/memory/memory%s/removable' % mem_block):
+        return True
+    return False
+
+
+def memory_hot_plug(mem_block):
+    """
+    Online the memory for the given block id.
 
+    :param mem_block: memory block id.
+    :type string: like 198
+    """
+    if _is_hot_pluggable(mem_block):
+        with open('/sys/devices/system/memory/memory%s/state' % mem_block, 'w') as fd:
+            fd.write('online')
+        if not _check_memory_state(mem_block):
+            raise ValueError("unable to online memory block, device busy?")
```

Comment:
According to the documentation: "exception ValueError: Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError."
I don't see a better builtin exception though. You can create your own exceptions here, like `MemoryStateError`, `MemoryHotPlugError`, ..., or even only one `MemoryError` that you can call providing the error message, like: 

```python
class MemoryError(Exception):
    """
    Called when some memory related operation fails
    """
    pass

...
...
def hotplug(block):
    if not _is_hot_pluggable(block):
        raise MemoryError("Memory block %s does not support hot-plug." % block)

    with open('/sys/devices/system/memory/memory%s/state' % mem_block, 'w') as state_file:
        state_file.write('online')

    if not _check_memory_state(block):
        raise MemoryError("Unable to hot-plug memory block %s. Device busy?" % block)
```